velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -15,178 +15,1541 @@
  */
 #include <fmt/format.h>
 
+#include "velox/dwio/dwrf/test/utils/BatchMaker.h"
 #include "velox/exec/tests/utils/PlanBuilder.h"
 #include "velox/functions/prestosql/aggregates/AggregateNames.h"
 #include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
 
 using facebook::velox::exec::test::PlanBuilder;
+using namespace facebook::velox::exec::test;
 
 namespace facebook::velox::aggregate::test {
 
 namespace {
 
+using facebook::velox::exec::test::MaterializedRow;
+
 class MinMaxByAggregationTest : public AggregationTestBase {
  protected:
   void testGlobalAggregation(
       const std::vector<RowVectorPtr>& vectors,
       const std::string& aggName,
       const std::string& colName) {
-    std::string funcName = aggName == kMaxBy ? "max" : "min";
-
-    for (int i = 0; i < 6; ++i) {
-      auto valColName = fmt::format("c{}", i);
-
+    const std::string funcName = aggName == kMaxBy ? "max" : "min";
+    for (int i = 0; i < rowType_->size(); ++i) {
+      const auto valColName = fmt::format("c{}", i);
+      const std::string verifyDuckDbSql = "SELECT " + valColName +
+          " FROM tmp WHERE " + colName + " = ( SELECT " + funcName + "(" +
+          colName + ") FROM tmp) LIMIT 1";
+      SCOPED_TRACE(fmt::format(
```

Comment:
I didn't know about SCOPED_TRACE. I read about it last night and it sounds very useful.